pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -231,7 +231,11 @@ func (r *hibernationReconciler) Reconcile(ctx context.Context, request reconcile
 	clusterSync := &hiveintv1alpha1.ClusterSync{}
 	if err := r.Get(context.Background(), types.NamespacedName{Namespace: cd.Namespace, Name: cd.Name}, clusterSync); err != nil {
 		// This may be NotFound, which means the clustersync controller hasn't created the ClusterSync yet.
-		// Fail and requeue to wait for it to exist.
+		// It does so shortly after the cluster is marked as installed. Requeue immediately to "poll" for it.
+		if apierrors.IsNotFound(err) {
+			cdLog.Info("clustersync does not exist yet; requeueing")
+			return reconcile.Result{Requeue: true}, nil
```

Comment:
Should be include a time-check here? ClusterSync controller will throw the error, but this controller will continuously requeue until someone fixes it - so how about we requeue if it has been less than 10 min since cd was created, or error out?

Reply:
I think I understand: you're talking about a scenario where the clustersync controller *can't* create the ClusterSync for some reason, so this would thrash indefinitely? That's a good thought; I was only thinking about the brief race condition right after the cluster finishes installing.

However, I think we're probably okay. The "immediate" requeue is still [subject to rate limiting](https://github.com/openshift/hive/blob/5b4fe1dce2677d80c0a8c1237eb909aeaa74b740/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go#L328) so it'll back off.

Unless there's some compelling reason to do otherwise, I'd like to leave that logic upstream and avoid trying to "outsmart" it here. Thoughts?